src/protocol/step.rs

```diff
@@ -0,0 +1,6 @@
+use super::Step;
```

Comment:
Want to move all the step related structs/traits here?

Reply:
Yes, will move them in the next PR.